lib/utils/import/multiple/index.ts

```diff
@@ -33,81 +30,45 @@ class MultipleImporter extends EventEmitter {
         file.name;
 
-      if (fileExtension === 'md' || fileExtension === 'txt') {
-        this.textFiles.push(file);
-      } else if (fileExtension === 'json') {
-        this.simplenoteFiles.push(file);
-      } else if (fileExtension === 'enex') {
-        this.evernoteFiles.push(file);
-      }
+      // Get the required importer based on the current files extension
+      // Add the file to this importers list of files
+      const theImporter = importers.find((importer) => {
+        return importer.fileTypes.indexOf(fileExtension) !== -1;
+      });
+      const currentFiles = this.importFiles.get(theImporter.name);
+      currentFiles.push(file);
+      this.importFiles.set(theImporter.name, currentFiles);
     }
+
+    let importersCompleted = 0;
     let totalImporters = 0;
-    let totalComplete = 0;
-    if (this.textFiles.length > 0) {
-      totalImporters++;
-    }
-    if (this.simplenoteFiles.length > 0) {
-      totalImporters++;
-    }
-    if (this.evernoteFiles.length > 0) {
-      totalImporters++;
-    }
 
-    if (this.textFiles.length > 0) {
-      const textImporter = new TextImporter(this.addNote, this.options);
-      textImporter.on('status', (type, arg) => {
-        if (type === 'complete') {
-          totalComplete++;
-          importedNoteCount += arg;
-          if (totalImporters === totalComplete) {
-            this.emit('status', 'complete', importedNoteCount);
-          }
-        } else {
-          this.emit('status', type, arg);
-        }
-      });
-      textImporter.importNotes(this.textFiles);
-    }
-    if (this.simplenoteFiles.length > 0) {
-      const simplenoteImporter = new SimplenoteImporter(
-        this.addNote,
-        this.options
-      );
-      simplenoteImporter.on('status', (type, arg) => {
-        if (type === 'complete') {
-          totalComplete++;
-          importedNoteCount += arg;
-          if (totalImporters === totalComplete) {
-            this.emit('status', 'complete', importedNoteCount);
-          }
-        } else {
-          this.emit('status', type, arg);
-        }
-      });
-      simplenoteImporter.importNotes(this.simplenoteFiles);
-    }
-    if (this.evernoteFiles.length > 0) {
-      const evernoteImporter = new EvernoteImporter(this.addNote, this.options);
-      evernoteImporter.on('status', (type, arg) => {
-        if (type === 'complete') {
-          totalComplete++;
-          importedNoteCount += arg;
-          if (totalImporters === totalComplete) {
-            this.emit('status', 'complete', importedNoteCount);
+    // Loop through each of the files to import and process them with the appropriate importer
+    this.importFiles.forEach(async (files, name) => {
+      if (files.length > 0) {
+        totalImporters++;
+        const { default: Importer } = await import(
+          /* webpackChunkName: 'utils-import-[request]' */ `../../../utils/import/${name}`
+        );
```

Comment:
this probably doesn't do what you think it does. it would be nice to move this into a function in the new `importers` module you created where the import paths can be explicit and static.

```ts
export const load = (name: ImporterName): Promise<???> => {
	switch (name) {
		case 'simplenote':
			return import(/* webpackChunkName: 'utils-import-simplenote' */ '../../../utils/import/simplenote');

		case 'evernote':
			return import(/* webpackChunkName: 'utils-import-evernote' */ '../../../utils/import/evernote');

		case 'text-files':
			return import(/* webpackChunkName: 'utils-import-text-files' */ '../../../utils/import/text-files');
	}

	throw new ValueError(`Unrecognized importer named ${name}`);
}
```

```ts
const { default: Importer } = await importers.load(name);
```

This allows us to find those import paths by string search and will prevent web pack from doing what it does, which I think would be to create a chunk/bundle for every file in the `utils/import/` path, possibly including the README. It additionally protects us against dynamic names exposing things we don't expect.

Reply:
You're right, that's not what I expected. I thought using the `[request]` would use the variable passed to the path based on examples I looked at. Sure seems you're right though looking at it again. Thank you!

Reply:
it might actually interpolate that `request`, but my point was more on the side of web pack preemptively creating chunks that we don't expect, because it uses some code analysis and tries to guess "what are the possible files that could be referenced by a variable in that path?" and that boils down to something like `../../../utils/import/*`